src/screens/WaitTimeScreen.tsx

```diff
@@ -1,42 +1,471 @@
-import { StatusBar } from "expo-status-bar";
-import { StyleSheet, View } from "react-native";
+import { useCallback, useContext, useState } from "react";
+import {
+  StyleSheet,
+  ScrollView,
+  View,
+  TouchableWithoutFeedback,
+  TouchableOpacity,
+  RefreshControl,
+} from "react-native";
 
-import { Button, WhiteSpace } from "@ant-design/react-native";
+import {
+  Button,
+  SearchBar,
+  Tag,
+  Modal,
+  Steps,
+  List,
+} from "@ant-design/react-native";
+import {
+  Feather,
+  MaterialIcons,
+  AntDesign,
+  Ionicons,
+} from "@expo/vector-icons";
 
 import { StyledText } from "@components/StyledText";
 import { LOCATION_DETAILS } from "@constants/routes";
-import { WaitTimeScreenProps } from "@navigators/WaitTimeStackNavigator";
+import { LocationDetailsScreenProps } from "@navigators/WaitTimeStackNavigator";
+import { ThemeContext } from "@contexts/theme";
+
+const Step = Steps.Step;
+const Item = List.Item;
+const Brief = Item.Brief;
+
+function renderIcon(type: any, theme: string): JSX.Element {
+  let color = "black";
+  if (theme === "dark") {
+    color = "white";
+  }
+  if (type === "food") {
+    return <MaterialIcons name={"restaurant"} size={24} color={color} />;
+  } else if (type === "shopping") {
+    return <AntDesign name="shoppingcart" size={24} color={color} />;
+  } else if (type === "transport") {
+    return <Ionicons name="bus-outline" size={24} color={color} />;
+  } else {
+    return <MaterialIcons name={"restaurant"} size={24} color={color} />; // just temporarily for the sake of return type
+  }
+}
+
+function renderLastUpdated(lastUpdated: number): string {
+  let string = "Last updated ";
+  if (lastUpdated > 60) {
+    let hours = Math.floor(lastUpdated / 60);
+    if (hours > 1) {
+      return string + hours + " hours ago";
+    } else {
+      return string + hours + " hour ago";
+    }
+  } else if (lastUpdated > 1) {
+    return string + lastUpdated + " mins ago";
+  } else if (lastUpdated === 1) {
+    return string + lastUpdated + " min ago";
+  } else {
+    return string + "now";
+  }
+}
 
 export const WaitTimeScreen = ({ navigation }: IWaitTimeScreenProps) => {
+  const { theme } = useContext(ThemeContext);
+  const [searchValue, setSearchValue] = useState("");
+  const [isVisible, setIsVisible] = useState(false);
+  const [sortBy, setSortBy] = useState("Distance");
```

Comment:
Can you make the options for sorting an enum instead of strings. Makes this feel a little safer.